python/GafferSceneUI/PathFilterUI.py

```diff
@@ -135,55 +170,47 @@ def __dataMenu( self, vectorDataWidget, menuDefinition ) :
 
 		selectedIndices = vectorDataWidget.selectedIndices()
 
+		filterData = vectorDataWidget.getData()[0]
+		selection = IECore.PathMatcher( [ filterData[row] for column, row in selectedIndices ] )
+
 		menuDefinition.append( "/selectDivider", { "divider" : True } )
 		menuDefinition.append(
 			"/Select Affected Objects",
 			{
-				"command" : functools.partial( Gaffer.WeakMethod( self.__selectAffected ), selectedIndices ),
+				"command" : functools.partial( _selectAffected, self.getPlug(), selection ),
 				"active" : len( selectedIndices ) > 0,
 			}
 		)
 
-	def __targetPlug( self, plug ) :
-
-		cellPlug = plug.ancestor( Gaffer.Spreadsheet.CellPlug )
-		if cellPlug is not None :
-			plug = plug.node()["out"][cellPlug.getName()]
-
-		stack = deque( plug.outputs() )
-
-		while stack :
-			plug = stack.popleft()
-			if isinstance( plug.node(), GafferScene.Filter ) :
-				return plug
-
-			stack.extend( plug.outputs() )
-
-		return None
+##########################################################################
+# Popup menu for Spreadsheet cells
+##########################################################################
 
-	def __selectAffected( self, selectedIndices ) :
+def __popupMenu( menuDefinition, plugValueWidget ) :
 
-		filterPlug = self.__targetPlug( self.getPlug() )
-		if filterPlug is None :
-			return
-		filterNode = filterPlug.node()
+	if not isinstance( plugValueWidget, GafferUI.VectorDataPlugValueWidget ) :
+		return
 
-		filterData = self.vectorDataWidget().getData()[0]
+	plug = plugValueWidget.getPlug()
+	if plug is None:
+		return
+	
+	cellPlug = plug.ancestor( Gaffer.Spreadsheet.CellPlug )
 
-		pathMatcher = IECore.PathMatcher()
-		for column, row in selectedIndices :
-			pathMatcher.addPath( filterData[row] )
+	if cellPlug is None :
+		return
```

Comment:
This isn't specific enough - it is putting "Select Affected Objects" on any `*VectorData` plug in a Spreadsheet (including IntVectorData!). How about we use `__targetPlug()` here, and tighten up `targetPlug()` to require that we've hit `PathFilter.paths`?

Maybe it would also make sense to rename `__targetPlug()` to `__pathsPlug()`?

Reply:
Fixed in 0e600279340949abc3f398908402ff2c4ebc8aa4